HerPublicWebsite.ManagementShell/FakeReferralGenerator.cs

```diff
@@ -0,0 +1,148 @@
+using HerPublicWebsite.BusinessLogic.Models.Enums;
+
+namespace HerPublicWebsite.ManagementShell;
+
+public interface IFakeReferralGenerator
+{
+    IEnumerable<ReferralRequest> GenerateFakeReferralRequests(int count, int startId);
+}
+
+public class FakeReferralGenerator : IFakeReferralGenerator
+{
+    public IEnumerable<ReferralRequest> GenerateFakeReferralRequests(int count, int startId)
+    {
+        var id = startId;
+        var custodianCodes = LocalAuthorityData.LocalAuthorityDetailsByCustodianCode.Keys;
+
+        var referralsSchema = new Faker<ReferralRequest>("en_GB")
+            .RuleFor(rr => rr.Id, _ => id++)
+            // Number, Street name
+            // Street name
+            .RuleFor(rr => rr.AddressLine1, f => f.Address.StreetAddress())
+            // 40% blank, 60% a second line of address
+            .RuleFor(rr => rr.AddressLine2, f => f.Address.City().OrNull(f, 0.4f))
+            .RuleFor(rr => rr.AddressTown, f => f.Address.City())
+            // NULL for almost all referrals
+            .RuleFor(rr => rr.AddressCounty, f => null)
+            .RuleFor(rr => rr.AddressPostcode, f => f.Address.ZipCode())
+            .RuleFor(rr => rr.CustodianCode, f => f.PickRandom<string>(custodianCodes))
+            // 8, 11 or 12 digits observed
+            .RuleFor(rr => rr.Uprn, f => f.Random.Long(10000000, 999999999999).ToString())
+            // Unknown 40%
+            // When it is known
+            //   A 1%
+            //   B 1%
+            //   C 1%
+            //   D 25%
+            //   E 25%
+            //   F 15%
+            //   G 7%
+            //   Expired 25%
+            .RuleFor(rr => rr.EpcRating, f => f.Random.WeightedRandom(
+                new[]
+                {
+                    EpcRating.A,
+                    EpcRating.B,
+                    EpcRating.C,
+                    EpcRating.D,
+                    EpcRating.E,
+                    EpcRating.F,
+                    EpcRating.G,
+                    EpcRating.Expired
+                },
+                new[] { 0.01f, 0.01f, 0.01f, 0.25f, 0.25f, 0.15f, 0.07f, 0.25f }))
+            // true 25%
+            .RuleFor(rr => rr.IsLsoaProperty, f => f.Random.Bool(0.25f))
+            // No 100%
+            .RuleFor(rr => rr.HasGasBoiler, f => HasGasBoiler.No)
+            // < 31000 40% 
+            // > 31000 4%
+            // < 34500 1%
+            // > 34500 0%
+            // < 36000 50%
+            // > 36000 5%
+            .RuleFor(rr => rr.IncomeBand,
+                f => f.Random.WeightedRandom(
+                    new[]
+                    {
+                        IncomeBand.UnderOrEqualTo31000,
+                        IncomeBand.GreaterThan31000,
+                        IncomeBand.UnderOrEqualTo34500,
+                        IncomeBand.GreaterThan34500,
+                        IncomeBand.UnderOrEqualTo36000,
+                        IncomeBand.GreaterThan36000
+                    },
```

Comment:
You can disable these warnings: would be good to do that here

Reply:
done